convert2rhel/actions/is_loaded_kernel_latest.py

```diff
@@ -0,0 +1,165 @@
+        #   C2R     1671212820      3.10.0-1160.81.1.el7    updates
+        # We need the `C2R` identifier to be present on the line so we can know for
+        # sure that the line we are working with is a line that contains
+        # relevant repoquery information to our check, otherwise, we just log the
+        # information as debug and do nothing with it.
+        for line in repoquery_output.split("\n"):
+            if line.strip() and "C2R" in line:
+                _, build_time, latest_kernel, repoid = tuple(str(line).split("\t"))
+                packages.append((build_time, latest_kernel, repoid))
+            else:
+                # Mainly for debugging purposes to see what is happening if we got
+                # anything else that does not have the C2R identifier at the start
+                # of the line.
+                logger.debug("Got a line without the C2R identifier: %s" % line)
+
+        # If we don't have any packages, then something went wrong, we need to
+        # decide whether to bail out or output a warning (only if the user used the
+        # special environment variable for it.
+        if not packages:
+            unsupported_skip = os.environ.get("CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK", None)
+            if not unsupported_skip:
+                self.set_result(
+                    status="ERROR",
+                    error_id="KERNEL_CURRENCY_CHECK_FAIL",
+                    message=(
+                        "Could not find any %s from repositories to compare against the loaded kernel.\n"
+                        "Please, check if you have any vendor repositories enabled to proceed with the conversion.\n"
+                        "If you wish to ignore this message, set the environment variable "
+                        "'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' to 1." % package_to_check
+                    ),
+                )
+                return
+
+            logger.warning(
+                "Detected 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
+                "the %s comparison.\n"
+                "Beware, this could leave your system in a broken state. " % package_to_check
+            )
+            return
+
+        packages.sort(key=lambda x: x[0], reverse=True)
+        _, latest_kernel, repoid = packages[0]
+
+        uname_output, _ = run_subprocess(["uname", "-r"], print_output=False)
+        loaded_kernel = uname_output.rsplit(".", 1)[0]
+        # append the package name to loaded_kernel and latest_kernel so they can be properly processed by
+        # compare_package_versions()
+        latest_kernel_pkg = "%s-%s" % (package_to_check, latest_kernel)
+        loaded_kernel_pkg = "%s-%s" % (package_to_check, loaded_kernel)
+        try:
+            match = compare_package_versions(latest_kernel_pkg, loaded_kernel_pkg)
+        except ValueError as exc:
+            logger.debug(exc)
```

Comment:
Wondering about this case right here... Should we treat that as an error as well, instead of just logging it out as debug? 

I mean, for the `ValueError` to happen it's possible that we made a mistake in the code and didn't notice, or, we got a package with funny parameters in it... 

Not sure if we should go with setting this to be an error, what do you think @abadger?

Reply:
We talked about this in chat and tentatively think it should be an error.  If the user thinks the version i fine, they can use the environment variable to skip the check.  We've asked the rest of the team for their input just to make sure, though.